.github/workflows/ci-pipeline.yml

```diff
@@ -59,4 +59,43 @@ jobs:
       - name: Run Snyk to check for vulnerabilities
         uses: snyk/actions/python-3.8@master
         env:
-          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
\ No newline at end of file
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+
+  docker:
+    if: github.event_name == 'push' && github.ref_name == 'main'
+    needs: build
+    name: Build and publish to Docker
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Docker Login
+        uses: docker/login-action@v2.0.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_ACCESSTOKEN }}
+      
+      - name: Build and push Docker image
+        run: |
+          docker build --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest .
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest
+
```

Comment:
Minor but I'd be tempted to have this as two separate steps; it'll just make the feedback a shade easier to digest if you have an error - was the problem with building the image or pushing it?